sdk/compute/Microsoft.Azure.Management.Compute/tests/VMScaleSetTests/VMScaleSetNetworkProfileTests.cs

```diff
@@ -431,6 +431,90 @@ public void TestVMScaleSetWithPublicIPAndPublicIPPrefix()
+                    EnsureClientsInitialized(context);
+
+                    ImageReference imageRef = GetPlatformVMImage(useWindowsImage: true);
+                    var storageAccountOutput = CreateStorageAccount(rgName, storageAccountName);
+                    var vnetResponse = CreateVNETWithSubnets(rgName, 2);
+                    var vmssSubnet = vnetResponse.Subnets[1];
+
+                    var publicipConfiguration = new VirtualMachineScaleSetPublicIPAddressConfiguration();
+                    publicipConfiguration.Name = "pip1";
+                    publicipConfiguration.IdleTimeoutInMinutes = 10;
+                    publicipConfiguration.DnsSettings = new VirtualMachineScaleSetPublicIPAddressConfigurationDnsSettings();
+                    publicipConfiguration.DnsSettings.DomainNameLabel = dnsname;
+                    publicipConfiguration.DeleteOption = DeleteOptions.Detach.ToString();
+
+                    VirtualMachineScaleSet vmScaleSet = CreateVMScaleSet_NoAsyncTracking(
+                        rgName: rgName,
+                        vmssName: vmssName,
+                        storageAccount: null,
+                        imageRef: imageRef,
+                        inputVMScaleSet: out inputVMScaleSet,
+                        singlePlacementGroup: false,
+                        createWithManagedDisks: true,
+                        vmScaleSetCustomizer:
+                            (virtualMachineScaleSet) => {
+                                virtualMachineScaleSet.UpgradePolicy = null;
+                                virtualMachineScaleSet.Overprovision = null;
+                                virtualMachineScaleSet.PlatformFaultDomainCount = 1;
+                                virtualMachineScaleSet.VirtualMachineProfile.NetworkProfile.NetworkApiVersion = NetworkApiVersion.TwoZeroTwoZeroHyphenMinusOneOneHyphenMinusZeroOne;
+                                virtualMachineScaleSet.VirtualMachineProfile.NetworkProfile
+                                    .NetworkInterfaceConfigurations[0].DeleteOption = DeleteOptions.Delete.ToString();
+                                virtualMachineScaleSet.VirtualMachineProfile.NetworkProfile
+                                    .NetworkInterfaceConfigurations[0].IpConfigurations[0].PublicIPAddressConfiguration = publicipConfiguration;
+                                virtualMachineScaleSet.OrchestrationMode = OrchestrationMode.Flexible.ToString();
+                                virtualMachineScaleSet.VirtualMachineProfile.StorageProfile.DataDisks = null;
+                            },
+                        createWithPublicIpAddress: false,
+                        subnet: vmssSubnet,
+                        capacity: 2);
+                    String vmssFilterMatch = $"'virtualMachineScaleSet/id' eq/subscriptions/{m_subId}/resourceGroups/{rgName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmssName}";
```

Comment:
This is incorrect formatting. You have to give space after eq and also include the resourceID in single quotes ('')
Format is :
'virtualMachineScaleSet/id' eq 'targetVmssResourceID'

I believe it should be 
$"'virtualMachineScaleSet/id' eq '/subscriptions/{m_subId}/resourceGroups/{rgName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmssName}'"

Reply:
Thanks Sachin, Got!!!